### DISCUSSION TITLE
models.di in Eclipse: initially locked; property data types are missing

### DISCUSSION THREAD

**Discussion by javac9** [openmdx/openmdx-example#6] [Q&A] [answered]
After project import, the classes and properties/attributes seem to be locked for editing. Only if I rename a class, it gets unlocked.
Additionally, the data types are missing for attributes. 

How do I fix these issues?

![Screenshot 2021-08-09 150144](https://user-images.githubusercontent.com/25393309/128710722-309d8650-bf69-494b-92a2-fac2ce693c9f.png)

**Comment by wfro64** [comment]
- Unlock the model to make it editable
- Add the openMDX models to the workspace for the openMDX-specific data types

Please ask Papyrus related questions [here](https://www.eclipse.org/forums/index.php/f/121/)

**Comment by javac9** [comment]
The issue with missing property data-types is resolved - I was importing the projects improperly.

But the issue of the diagram components being initially locked persists. I posted the question in the Papyrus forum as you suggested - one of the members contacted me over LinkedIn and is so far unable to help me determine why there is such a behaviour.

This is a screencast I made to demonstrate how it looks and how I manage to "unlock" the diagram (by renaming any class). You might have an idea what's wrong here.

Info about my environment (Eclipse Papyrus forum where I posted them): 
https://www.eclipse.org/forums/index.php/m/1843717/

The screencast demo:
https://drive.google.com/file/d/1_t6j_F_GI_dP7YDj8WMncpc6_U9ziyXT/view?usp=sharing

**Comment by wfro64** [comment]
You have to write enable the (EMF) model in the context menu of the Model Explorer. Write enabling diagram elements has no effect.

![Screenshot_20210816_131841](https://user-images.githubusercontent.com/11387061/129556458-aa466957-d677-4271-a913-be37efbe284c.png)

**Reply by javac9** [reply]
Just tried it, doesn't seem to help. See for yourself, I'm afraid something might be wrong with my setup although it appears normal:
https://drive.google.com/file/d/1lo5Bhh6c102dabH4L5u57i7H8tgThM8g/view?usp=sharing

**Comment by wfro64** [comment, marked as answer]
"Enable write" does obviously not work with some versions of Papyrus if the href Property in the `models.notation` files is not properly URL-encoded. Replaced as follows:

* `openMDX 2 Example ~ HelloWorld (EMF)` → `openMDX%202%20Example%20~%20HelloWorld%20(EMF)`
* `openMDX 2 Example ~ Workshop (EMF)` → `openMDX%202%20Example%20~%20Workshop%20(EMF)`

Please pull and try again.

**Reply by javac9** [reply]
Yes, it works now, thanks! However, you don't have to "Enable write" from the Model explorer, you can do it also by clicking on any diagram component.
